api/v1alpha1/appgroup_types.go

```diff
@@ -62,11 +62,29 @@ type ApplicationSpec struct {
 	// +required
 	Release *Release `json:"release"`
 
+	// Executor holds container customizations for the executor to use when running pods
+	// +optional
+	Executor *Executor `json:"executor,omitempty"`
+
 	// Subcharts provides the dependency order among the subcharts of the application
 	// +optional
 	Subcharts []DAG `json:"subcharts,omitempty"`
 }
 
+type Executor struct {
```

Comment:
@nitishm  Let me try and understand the scenario here, is the idea that someone will customize the executor/docker image per-application group? What is the viable scenario for this? Can we not use a single docker image and then dynamically adjust the checks we do within that executable?